packages/gatsby-plugin-offline/sw-append.js

```diff
@@ -12,22 +12,7 @@ const navigationRoute = new workbox.routing.NavigationRoute(({ event }) => {
       const offlineShell = `%pathPrefix%/offline-plugin-app-shell-fallback/index.html`
       const cacheName = workbox.core.cacheNames.precache
 
-      return caches.match(offlineShell, { cacheName }).then(cachedResponse => {
-        if (!cachedResponse) {
-          return fetch(offlineShell).then(response => {
-            if (response.ok) {
-              return caches.open(cacheName).then(cache =>
-                // Clone is needed because put() consumes the response body.
-                cache.put(offlineShell, response.clone()).then(() => response)
-              )
-            } else {
-              return fetch(event.request)
-            }
-          })
-        }
-
-        return cachedResponse
-      })
+      return caches.match(offlineShell, { cacheName })
```

Comment:
I don't think it's safe to remove that yet. This handles stuff before runtime actually run, so we can't really rely that any changes we make in runtime are actually in effect (not to mention that people can use "missmatched" versions of gatsby and offline plugin in this could "brick" their sites still in those still not figured out scenarios)